server/datastore/mysql/mdm.go

```diff
@@ -2275,3 +2286,131 @@ SELECT EXISTS (
 	}
 	return exists, nil
 }
+
+type batchSetAssociationCurrentProfile struct {
+	ProfileUUID string `db:"profile_uuid"`
+	Name        string `db:"name"`
+}
+
+type BatchSetAssociationIncomingProfile struct {
+	Name             string // Identifier or name depending on Platform.
+	ProfileUUID      string
+	LabelsIncludeAll []fleet.ConfigurationProfileLabel
+	LabelsIncludeAny []fleet.ConfigurationProfileLabel
+	LabelsExcludeAny []fleet.ConfigurationProfileLabel
+}
+
+// batchSetAssociations handles setting label and variable associations for all types of platform profiles.
+func (ds *Datastore) batchSetLabelAndVariableAssociations(ctx context.Context, tx sqlx.ExtContext, platform string, teamID *uint, incomingProfiles []*BatchSetAssociationIncomingProfile, profilesVariablesByIdentifier []fleet.MDMProfileIdentifierFleetVariables) (updatedLabels bool, err error) {
+	if len(incomingProfiles) == 0 {
+		return false, nil // Do nothing on empty incoming profiles.
+	}
+
+	var currentProfilesQuery string
+
+	// Set platform specific variables here to use later.
+	switch platform {
+	case "darwin":
+		currentProfilesQuery = `SELECT profile_uuid, identifier as name FROM mdm_apple_configuration_profiles WHERE team_id = ? AND identifier IN (?)`
```

Comment:
Just to double-check that we didn't remove anything, this was not supported in declarations?

Reply:
We did not declarations uses it's own flow, which is most likely why I missed it.

Looking at it right now, declarations does not support variables, so the shared method, would need a bit of rework to reliably go around that for declarations and a lot of mapping depending on platform (+ type) for android.

So not sure if it's worth atm. to go further on this, than the shared in it's current state for now.